pkg/restic/common_test.go

```diff
@@ -0,0 +1,284 @@
+		t.Run(test.name, func(t *testing.T) {
+			pod := &corev1api.Pod{}
+			pod.Annotations = test.annotations
+
+			SetPodSnapshotAnnotation(pod, test.volumeName, test.snapshotID)
+			assert.EqualValues(t, test.expected, pod.Annotations)
+		})
+	}
+}
+
+func TestGetVolumesToBackup(t *testing.T) {
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		expected    []string
+	}{
+		{
+			name:        "nil annotations",
+			annotations: nil,
+			expected:    nil,
+		},
+		{
+			name:        "no volumes to backup",
+			annotations: map[string]string{"foo": "bar"},
+			expected:    nil,
+		},
+		{
+			name:        "one volume to backup",
+			annotations: map[string]string{"foo": "bar", volumesToBackupAnnotation: "volume-1"},
+			expected:    []string{"volume-1"},
+		},
+		{
+			name:        "multiple volumes to backup",
+			annotations: map[string]string{"foo": "bar", volumesToBackupAnnotation: "volume-1,volume-2,volume-3"},
+			expected:    []string{"volume-1", "volume-2", "volume-3"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pod := &corev1api.Pod{}
+			pod.Annotations = test.annotations
+
+			assert.EqualValues(t, test.expected, GetVolumesToBackup(pod))
+		})
+	}
+}
+
+func TestGetSnapshotsInBackup(t *testing.T) {
+	tests := []struct {
+		name             string
+		arkBackupName    string
+		podVolumeBackups []arkv1api.PodVolumeBackup
+		expected         []SnapshotIdentifier
+	}{
+		{
+			name:             "no pod volume backups",
+			arkBackupName:    "backup-1",
+			podVolumeBackups: nil,
+			expected:         nil,
+		},
+		{
+			name:          "no pod volume backups with matching label",
+			arkBackupName: "backup-1",
+			podVolumeBackups: []arkv1api.PodVolumeBackup{
+				{ObjectMeta: metav1.ObjectMeta{Name: "foo", Labels: map[string]string{arkv1api.BackupNameLabel: "non-matching-backup-1"}}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "bar", Labels: map[string]string{arkv1api.BackupNameLabel: "non-matching-backup-2"}}},
+			},
+			expected: nil,
+		},
+		{
+			name:          "some pod volume backups with matching label",
+			arkBackupName: "backup-1",
+			podVolumeBackups: []arkv1api.PodVolumeBackup{
+				{ObjectMeta: metav1.ObjectMeta{Name: "foo", Labels: map[string]string{arkv1api.BackupNameLabel: "non-matching-backup-1"}}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "bar", Labels: map[string]string{arkv1api.BackupNameLabel: "non-matching-backup-2"}}},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "pvb-1", Labels: map[string]string{arkv1api.BackupNameLabel: "backup-1"}},
+					Spec: arkv1api.PodVolumeBackupSpec{
+						Pod: corev1api.ObjectReference{Namespace: "ns-1"},
+					},
+					Status: arkv1api.PodVolumeBackupStatus{SnapshotID: "snap-1"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "pvb-2", Labels: map[string]string{arkv1api.BackupNameLabel: "backup-1"}},
+					Spec: arkv1api.PodVolumeBackupSpec{
+						Pod: corev1api.ObjectReference{Namespace: "ns-2"},
```

Comment:
Let's add the pod's name as it's invalid without it.

Reply:
done.